src/Features/Core/Portable/PullMemberUp/AbstractMembersPullerService.cs

```diff
@@ -277,14 +269,17 @@ private static ISymbol GetSymbolsToPullUp(MemberAnalysisResult analysisResult)
                         solution.GetDocumentId(syntax.SyntaxTree),
                         cancellationToken).ConfigureAwait(false);
 
+                    var semanticModel = await solution.GetDocument(solution.GetDocumentId(syntax.SyntaxTree)).
+                        GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
                     // add import symbol annotations for all the code we may be moving
                     newDestination = newDestination.WithAdditionalAnnotations(
-                        syntax.DescendantNodesAndSelf()
-                        .Select((node) => semanticModel.GetTypeInfo(node, cancellationToken).Type)
-                        .Where((type) => type != null &&
-                                !type.IsErrorType())
-                        .Distinct()
-                        .Select((type) => SymbolAnnotation.Create(type)));
+                        syntax.DescendantNodesAndSelf().
+                        Where((node) => FilterType(node)).
+                        Select((node) => semanticModel.GetTypeInfo(node, cancellationToken).Type).
+                        Where((type) => type != null).
+                        Distinct().
+                        Select((type) => SymbolAnnotation.Create(type)));
```

Comment:
```suggestion
                        Where(node => FilterType(node)).
                        Select(node => semanticModel.GetTypeInfo(node, cancellationToken).Type).
                        Where(type => type != null).
                        Distinct().
                        Select(type => SymbolAnnotation.Create(type)));
```

Reply:
yes.  please don't parenthesize single parameters.

Reply:
this will not work for extension methods.

Reply:
this may be an enormous number of annotations. i'm not comfortable with this.